hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/compact/strategy/DayBasedCompactionStrategy.java

```diff
@@ -63,21 +60,9 @@ public Comparator<String> getComparator() {
     return comparator;
   }
 
-  @Override
-  public List<HoodieCompactionOperation> orderAndFilter(HoodieWriteConfig writeConfig,
-      List<HoodieCompactionOperation> operations, List<HoodieCompactionPlan> pendingCompactionPlans) {
-    // Iterate through the operations and accept operations as long as we are within the configured target partitions
-    // limit
-    return operations.stream()
-        .collect(Collectors.groupingBy(HoodieCompactionOperation::getPartitionPath)).entrySet().stream()
-        .sorted(Map.Entry.comparingByKey(comparator)).limit(writeConfig.getTargetPartitionsPerDayBasedCompaction())
-        .flatMap(e -> e.getValue().stream()).collect(Collectors.toList());
-  }
-
   @Override
   public List<String> filterPartitionPaths(HoodieWriteConfig writeConfig, List<String> allPartitionPaths) {
-    return allPartitionPaths.stream().map(partition -> partition.replace("/", "-"))
-        .sorted(Comparator.reverseOrder()).map(partitionPath -> partitionPath.replace("-", "/"))
+    return allPartitionPaths.stream().sorted(comparator)
         .collect(Collectors.toList()).subList(0, Math.min(allPartitionPaths.size(),
```

Comment:
Seems a pure code optimization?

Reply:
Can you check the test failures.

Reply:
The test failures are not related to this change.

Reply:
Why we subList its original size, I'm confused.

Reply:
If the original size `allPartitionPaths.size()` is smaller than `writeConfig.getTargetPartitionsPerDayBasedCompaction()`, only `subList(writeConfig.getTargetPartitionsPerDayBasedCompaction())` will throw IndexOutOfBoundsException. I think we can use `limit` to replace `subList()`.

Reply:
Okay, got it, caution that you have change the compactor of the partitions, does that introduce any protential regressions ? Can we add some tests to conver it.

Reply:
Unit tests updated

Reply:
Is the partitiona path relative or full, why it has code like this before:

```java
partition.replace("/", "-")
```

Reply:
The partition path is relative path.

In before code, `replace` is just used to sort the partitions by dictionary order. If we use the date format compator, no need to replace it.
```
allPartitionPaths.stream().map(partition -> partition.replace("/", "-"))
        .sorted(Comparator.reverseOrder()).map(partitionPath -> partitionPath.replace("-", "/"))
```